File: dotnet/src/Core/Models/Entities/Credential.cs
using System.Text.Json.Serialization;
using Agience.Core.Models.Entities.Abstract;
using Agience.Core.Models.Enums;

namespace Agience.Core.Models.Entities
{
    public class Credential : BaseEntity
    {

        [JsonPropertyName("agent_id")]
        public string? AgentId { get; set; }

        [JsonPropertyName("connection_id")]
        public string? ConnectionId { get; set; }

        [JsonPropertyName("authorizer_id")]
        public string? AuthorizerId { get; set; }

        [JsonPropertyName("status")]
        public CredentialStatus Status { get; set; } = CredentialStatus.NoAuthorizer;

        [JsonPropertyName("refresh_token")]
        public string? RefreshToken { get; set; }

        [JsonPropertyName("access_token")]
        public string? AccessToken { get; set; }

        [JsonPropertyName("expiry_date")]
        public DateTime? ExpiryDate { get; set; }

    }
}

File: python/src/core/models/entities/credential.py
from pydantic import Field
from typing import Optional
from datetime import datetime
from core.models.entities.abstract.base_entity import BaseEntity
from core.models.enums.enums import CredentialStatus


class Credential(BaseEntity):
    agent_id: Optional[str] = Field(None, alias="agent_id")
    connection_id: Optional[str] = Field(None, alias="connection_id")
    authorizer_id: Optional[str] = Field(None, alias="authorizer_id")
    status: CredentialStatus = Field(
        default=CredentialStatus.NoAuthorizer, alias="status")
    refresh_token: Optional[str] = Field(None, alias="refresh_token")
    access_token: Optional[str] = Field(None, alias="access_token")
    expiry_date: Optional[datetime] = Field(None, alias="expiry_date")

    class Config:
        allow_population_by_field_name = True
        use_enum_values = True
